services/src/main/java/org/keycloak/services/resources/admin/ComponentResource.java

```diff
@@ -160,32 +165,39 @@ public ComponentRepresentation getComponent(@PathParam("id") String id) {
     @Consumes(MediaType.APPLICATION_JSON)
     public Response updateComponent(@PathParam("id") String id, ComponentRepresentation rep) {
         auth.realm().requireManageRealm();
```

Comment:
Yeah, it is bound to the external session, so I figured it would be better to leave it as is, outside of the new tx. On the flip side, if the authorization check fails it won't start a new transaction, so there's a little gain there :D

Reply:
ok, resolved for me.